ios/RNSScreen.h

```diff
@@ -124,6 +125,7 @@ namespace react = facebook::react;
 - (void)updateBounds;
 - (void)notifyDismissedWithCount:(int)dismissCount;
 - (instancetype)initWithFrame:(CGRect)frame;
+- (void)onWillReceiveRemoveMutation;
```

Comment:
```suggestion
/// Tell `Screen` that it will be unmounted in next transaction.
/// The component needs this so that we can later decide whether to 
/// replace it with snapshot or not. 
- (void)willBeUnmountedInUpcomingTransaction
```

I know I've suggested the current name, but when I'm thinking of it now: the Screen won't receive mutation, it will be only subject of one.